src/Nri/Ui/Util.elm

```diff
@@ -26,3 +26,22 @@ removePunctuation =
                 |> Maybe.withDefault Regex.never
     in
     Regex.replace regex (always "")
+
+
+{-| Creates a lowercased string that is safe to use for HTML IDs.
+-}
+safeIdString : String -> String
+safeIdString =
+    let
+        unsafeChar =
+            Regex.fromString "(^[^a-zA-Z]+|[^a-zA-Z0-9_-]+)" |> Maybe.withDefault Regex.never
```

Comment:
I think we need to keep numbers, no? 

Could we use replacePunctuation here?

Reply:
Ohhhh I think I was reading the regex wrong

Reply:
I am curious though -- could we use removePunctuation?

Reply:
I've tagged @ap-nri as he is the regex mastermind behind this =D. We were messing around with this one day during elm bookclub and this was the implementation that we landed on to solve this problem somewhere in monolith. I must confess Regex might as well be hieroglyphics to me so I hope Alex can further elaborate/remember what we talked about in bookclub.

Reply:
`removePunctuation` was written wrong. The char range `A-z` starts at `A` and goes to `z` including all of the punctuation marks between `Z` and `a` which are, in order with some context around them:
```
XYZ[\]^_`abc
```

Reply:
see my comments in the code for more of how this works.